codeSnippets/snippets/client-call-id/src/main/kotlin/com/example/Application.kt

```diff
@@ -0,0 +1,39 @@
+package com.example
+
+import io.ktor.client.*
+import io.ktor.client.engine.cio.*
+import io.ktor.client.request.*
+import io.ktor.client.statement.*
+import io.ktor.client.plugins.callid.*
+import io.ktor.http.*
+import kotlinx.coroutines.*
+
+fun main() {
+    runBlocking {
+        val client = HttpClient(CIO) {
```

Comment:
I would extract every plugin usage to a separate function with a specific name

Reply:
Do you mean like this:

```kotlin
fun main() {
        getRequestWithCustomCallId()
        //...
}

fun getRequestWithCustomCallId() {
    runBlocking {
        val client = HttpClient(CIO) {
            install(CallId) {
                generate { "call-id-client" }
                addToHeader(HttpHeaders.XRequestId)
                intercept { request, callId -> request.parameter("callId", callId) }
            }
        }
        val response: HttpResponse = client.get("http://0.0.0.0:8090/call")
        println(response.bodyAsText())

        val responseFromDoubleCall: HttpResponse = client.get("http://0.0.0.0:8090/double-call")
        println(responseFromDoubleCall.bodyAsText())
        client.use {
            client.close()
        }
    }
}
//...
```

Reply:
fun Closeable.use(block: () -> Unit) {
}